### PR TITLE
drop table clear cache add if exists support, like "drop table if exists test1"

### PR DESCRIPTION
**Pull request by jilieryuyi**
match is exists support

### CODE REVIEW

File: canal/sync.go [hunk 1]
```diff
@@ -128,7 +127,7 @@ func (c *Canal) runSyncBinlog() error {
 				trigger = true
 			} else if mb = checkDropTable(e); mb != nil {
 				schema = e.Schema
-				table = mb[2]
+				table = mb[7]
```
Comment (siddontang) [inline canal/sync.go]: I am confused that why 7 here now?

Comment (jilieryuyi) [reply canal/sync.go]: yes, you can find this test in canal_test.go , the test function is : TestDropTableExp

**Comment by jilieryuyi** [comment]
@siddontang optimize with 3 now！

File: canal/sync.go [hunk 2]
```diff
@@ -15,9 +15,9 @@ import (
 
 var (
 	expAlterTable  = regexp.MustCompile("(?i)^ALTER\\sTABLE\\s.*?`{0,1}(.*?)`{0,1}\\.{0,1}`{0,1}([^`\\.]+?)`{0,1}\\s.*")
-	//rename should clear old table cache
 	expRenameTable = regexp.MustCompile("(?i)^RENAME\\sTABLE\\s.*?`{0,1}(.*?)`{0,1}\\.{0,1}`{0,1}([^`\\.]+?)`{0,1}\\s{1,}TO\\s.*?")
-	expDropTable   = regexp.MustCompile("(?i)^DROP\\sTABLE\\s.*?`{0,1}(.*?)`{0,1}\\.{0,1}`{0,1}([^`\\.]+?)`{0,1}($|\\s)")
+	//add if exists support
```
Comment (siddontang) [inline canal/sync.go]: what does the comment mean?

Comment (jilieryuyi) [reply canal/sync.go]: This is just a line of notes

**Comment by siddontang** [comment]
Thanks @jilieryuyi

**Pull request merged**